### PR TITLE
docs: update quick start guide

### PR DESCRIPTION
**Pull request by shabeebk**
Fix grammar to have a better reading experience.

Change `deployment/argo-server` to `services/argo-server` in port forwarding instructions.

Fixes #TODO

### Motivation

I was reading through the quick start guide and the giscus comments on the page. Found that `kubectl -n argo port-forward deployment/argo-server 2746:2746` can be changed to `kubectl -n argo port-forward services/argo-server 2746:2746`. While changing it, I have seen some grammar changes that could improve the readability of the page.

### Modifications

Quick start guide changed.

### CODE REVIEW

**Review by PranitRout07 — COMMENTED** [review]
LGTM

File: docs/quick-start.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 To see how Argo Workflows work, you can install it and run examples of simple workflows.
 
-Before you start you need a Kubernetes cluster and `kubectl` set up to be able to access that cluster. For the purposes of getting up and running, a local cluster is fine. You could consider the following local Kubernetes cluster options:
+Before you start you need a Kubernetes cluster and `kubectl` set up to be able to access that cluster. To get up and run the argo workflow, a local cluster is fine. You could consider the following local Kubernetes cluster options:
```
Comment (jmeridth) [inline docs/quick-start.md:5]: I don't think this reads better.  The prior text is clearer and worded better IMO.

File: docs/quick-start.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ argo logs -n argo @latest
 * Open a port-forward so you can access the UI:
 
 ```bash
-kubectl -n argo port-forward deployment/argo-server 2746:2746
+kubectl -n argo port-forward services/argo-server 2746:2746
```
Comment (jmeridth) [inline docs/quick-start.md:103]: I think you can port forward either the deployment or the service, but we want to do the service it is usually singular (e.g., `... port-forward service/argo-server`).

**Review by jmeridth — COMMENTED** [review]
A couple notes.  Otherwise good.

File: docs/quick-start.md (hunk 2 above)
Comment (shabeebk) [reply docs/quick-start.md]: Thanks for that. Updated.

File: docs/quick-start.md (hunk 1 above)
Comment (shabeebk) [reply docs/quick-start.md]: Thanks for that. Reverted.

**Review by jmeridth — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]

File: docs/quick-start.md (hunk 1 above)
Comment (agilgur5) [reply docs/quick-start.md]: Agreed that the prior text was better. Technically speaking, they both actually fail the k8s style guide on [avoiding jargon](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-jargon-and-idioms). I might say "For quick start purposes" to match the existing language / title of the page.

Some of the docs I have yet to touch in general could use a bit of simplification ([per k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)).

Comment (agilgur5) [reply docs/quick-start.md]: Made some significant simplifications as mentioned above in #12772

**Pull request merged**